protocol/src/main/java/org/geysermc/mcprotocollib/network/tcp/TcpSession.java

```diff
@@ -188,31 +187,21 @@ public void callPacketSent(Packet packet) {
     }
 
     @Override
-    public int getCompressionThreshold() {
-        return this.compressionThreshold;
-    }
-
-    @Override
-    public void setCompressionThreshold(int threshold, boolean validateDecompression) {
-        this.compressionThreshold = threshold;
-        if (this.channel != null) {
-            if (this.compressionThreshold >= 0) {
-                if (this.channel.pipeline().get("compression") == null) {
-                    this.channel.pipeline().addBefore("codec", "compression",
-                        new TcpPacketCompression(this, new ZlibCompression(), validateDecompression));
-                }
-            } else if (this.channel.pipeline().get("compression") != null) {
-                this.channel.pipeline().remove("compression");
-            }
+    public void setCompression(CompressionConfig compressionConfig) {
+        if (this.channel == null) {
+            throw new IllegalStateException("You need to be connected to set the compression!");
```

Comment:
```suggestion
            throw new IllegalStateException("You need to connect to set the compression!");
```
consistency with setEncryption?